src/Snapshot/SnapshotStore.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcing\Snapshot;
+
+use Patchlevel\EventSourcing\Aggregate\SnapshotableAggregateRoot;
+
+interface SnapshotStore
+{
+    public function save(Snapshot $snapshot): void;
+
+    /**
+     * @param class-string<SnapshotableAggregateRoot> $aggregate
+     */
+    public function load(string $aggregate, string $id): ?Snapshot;
```

Comment:
i would be more happy if the api would not contain a `null` case and instead throw an exception

Reply:
done